python/cudf/cudf/core/column/lists.py

```diff
@@ -176,6 +176,37 @@ def __init__(self, column, parent=None):
             )
         super().__init__(column=column, parent=parent)
 
+    def extract(self, index):
+        """
+        Extracts the element at the given index from each
+        row of a Series of lists
+
+        Parameters
+        ----------
+        index : int
+            position of the desired element in the list
+
+        Returns
+        -------
+        Series
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], [3, 4, 5], [4, 5, 6]])
+        >>> s.list.extract(-1)
+        0    3
+        1    5
+        2    6
+        dtype: int64
+        """
+        if abs(index) >= self.len().min():
+            # TODO: handle case when  index == -self.len().min()
+            # e.g: index= -3 should return [1,3,4]
+            # instead of IndexError
```

Comment:
Doesn't libcudf already support negative indices?

https://github.com/rapidsai/cudf/blob/c1c60ba3daf36d0ee5553558f70669f454d9f0c8/cpp/tests/lists/extract_tests.cpp#L83-L87

Reply:
The problem isn't with all negative indices  ( e.g: `index = -2` works). I need to handle case when `index == -min` specifically, which my current implementation considers to be out of bound in that `abs(-3)= 3 = min `

Reply:
Maybe something like:
```
min_col_list_len = self.len().min()
if (index >=0 and index < min_col_list_len) or (index < 0 and abs(index) <= min_col_list_len):
    ...
```